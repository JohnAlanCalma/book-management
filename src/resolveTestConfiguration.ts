// tslint:disable ordered-imports
import 'reflect-metadata'
import { serviceContainer } from '@/inversify/config'
import { stubProviders } from '@/inversify/resolveStubProviders'
import Vue from 'vue'
import Router from 'vue-router'
import Vuetify from 'vuetify'
import Vuex from 'vuex'

Vue.use(Router)
Vue.use(Vuex)

stubProviders(serviceContainer)

Vue.use(Vuetify, {
  theme: {
    primary: '#424242',
    secondary: '#FF8A80',
    accent: '#BA68C8',
    error: '#f44336',
    warning: '#ffeb3b',
    info: '#2196f3',
    success: '#4caf50',
  },
})
