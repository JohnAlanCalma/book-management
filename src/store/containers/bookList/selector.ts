import { Store } from '@/store/root'
import { ScreenState } from './index'

function _stateScope(state: Store['state']) {
  return state.containers.bookList
}

function isInitialized(state: Store['state']) {
  return _stateScope(state).screenState !== ScreenState.INITIAL_FETCHING
}

function books(state: Store['state']) {
  return Object
    .values(_stateScope(state).books)
    .sort((a, b) => a.updatedAt > b.updatedAt ? 1 : -1)
}

export default {
  isInitialized,
  books,
}
