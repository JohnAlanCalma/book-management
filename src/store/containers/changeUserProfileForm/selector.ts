import {Store} from '@/store/root'
import {ScreenState} from './index'

function _stateScope(state: Store['state']) {
  return state.containers.changeUserProfileForm
}

function isSending(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SENDING
}

function isSendFailed(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SEND_FAILED
}

function isSendSuccess(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SEND_SUCCESS
}

function isOpen(state: Store['state']) {
  return _stateScope(state).isOpen
}

export default {
  isSending,
  isSendSuccess,
  isSendFailed,
  isOpen,
}
