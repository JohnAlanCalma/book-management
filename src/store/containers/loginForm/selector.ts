import { Store } from '@/store/root'
import { ScreenState } from './index'

function _stateScope(state: Store['state']) {
  return state.containers.loginForm
}

function isSending(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SENDING
}

function isFailed(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.LOGIN_FAILED
}

export default {
  isSending,
  isFailed,
}
