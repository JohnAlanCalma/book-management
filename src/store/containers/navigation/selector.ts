import {Store} from '@/store/root'

function _stateScope(state: Store['state']) {
  return state.containers.navigation
}

function isOpen(state: Store['state']) {
  return _stateScope(state).isOpen
}

export default {
  isOpen,
}
