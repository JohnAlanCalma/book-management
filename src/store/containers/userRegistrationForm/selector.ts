import { Store } from '@/store/root'
import { ScreenState } from './index'

function _stateScope(state: Store['state']) {
  return state.containers.userRegistrationForm
}

function isSending(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SENDING
}

function isSendFailed(state: Store['state']) {
  return _stateScope(state).screenState === ScreenState.SEND_FAILED
}

function errorMessage(state: Store['state']) {
  switch (_stateScope(state).errorCode) {
    case 'auth/email-already-in-use':
      return 'そのメールアドレスは既に登録されています'
    case 'auth/invalid-email':
      return 'メールアドレスの形式が間違っています'
    case 'auth/operation-not-allowed':
      return 'システムでメールアドレス・パスワード認証が有効になっていません'
    case 'auth/weak-password':
      return '脆弱なパスワードです'
    default:
      return 'エラーが発生しました\n時間を置いてから再度やり直してください'
  }
}

export default {
  isSending,
  isSendFailed,
  errorMessage,
}
