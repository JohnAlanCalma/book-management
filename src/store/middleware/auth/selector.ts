import {Store} from '@/store/root'

function _stateScope(state: Store['state']) {
  return state.middleware.auth
}

function isInitialized(state: Store['state']) {
  return _stateScope(state).isInitialized
}

function isLoggedIn(state: Store['state']) {
  return _stateScope(state).isLoggedIn
}

function user(state: Store['state']) {
  return _stateScope(state).user
}

function isEmailVerified(state: Store['state']) {
  return _stateScope(state).isEmailVerified
}

export default {
  isInitialized,
  isLoggedIn,
  user,
  isEmailVerified,
}
