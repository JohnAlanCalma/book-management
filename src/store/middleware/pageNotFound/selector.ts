import {Store} from '@/store/root'

function _stateScope(state: Store['state']) {
  return state.middleware.pageNotFound
}

function isPageNotFound(state: Store['state']) {
  return _stateScope(state).isPageNotFound
}

export default {
  isPageNotFound,
}
