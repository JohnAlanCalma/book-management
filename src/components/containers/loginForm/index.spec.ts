import '@/resolveTestConfiguration'
import selector from '@/store/containers/loginForm/selector'
import authSelector from '@/store/middleware/auth/selector'
import { createStore } from '@/store/root'
import { shallowMount } from '@vue/test-utils'
import flushPromises from 'flush-promises'
import index from './index.vue'

test('login form container', async () => {
  const store = createStore()
  selector.isFailed(store.state)
  selector.isSending(store.state)
  const wrapper = shallowMount(index, { store })
  expect(wrapper.html()).toMatchSnapshot()
  expect(selector.isFailed(store.state)).toBe(false)
  expect(selector.isSending(store.state)).toBe(false)
  const vm = wrapper.vm as any
  vm.login({
    email: 'tanaka@gmail.com',
    password: 'tanakatarou',
  })
  expect(selector.isSending(store.state)).toBe(true)
  expect(authSelector.user(store.state)).toBeNull()
  await flushPromises()
  expect(authSelector.user(store.state)).not.toBeNull()
  expect(selector.isFailed(store.state)).toBe(false)
})
