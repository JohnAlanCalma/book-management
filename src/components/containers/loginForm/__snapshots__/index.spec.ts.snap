// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login form container 1`] = `
<vflex-stub tag="div" xs12="" sm8="" md6="">
  <loginform-stub login="function () { [native code] }"></loginform-stub>
  <vdialog-stub persistent="true" maxwidth="none" origin="center center" width="300" transition="dialog-transition">
    <vcard-stub color="primary" dark="">
      <vcardtext-stub>
        ログイン中です...
        <vprogresslinear-stub active="true" buffervalue="100" color="white" height="7" indeterminate="true" value="0" class="mb-0"></vprogresslinear-stub>
      </vcardtext-stub>
    </vcard-stub>
  </vdialog-stub>
  <vdialog-stub maxwidth="none" origin="center center" width="500" transition="dialog-transition">
    <vcard-stub color="error" dark="">
      <vcardtext-stub>ログイン情報が間違っています</vcardtext-stub>
      <vcardactions-stub>
        <vspacer-stub></vspacer-stub>
        <vbtn-stub activeclass="v-btn--active" tag="button" type="button" color="red lighten-2">OK</vbtn-stub>
      </vcardactions-stub>
    </vcard-stub>
  </vdialog-stub>
</vflex-stub>
`;
